examples/init.pp

```diff
@@ -12,15 +12,14 @@
 
 include caddy
 
-
-file {'/var/www':
+file { '/var/www':
   ensure => directory,
 }
 
-caddy::vhost {'example1':
-  source => 'puppet:///modules/caddy/etc/caddy/config/example1.conf',
+caddy::vhost { 'example1':
+  content => file('caddy/examples/example1.conf'),
```

Comment:
Given its size, wouldn't it be clearer to use heredoc and inline it in the example?

Reply:
This is pretty much useless, because we already have it at ```manifests/vhosts.pp``` and ```REFERENCE.md``` for what it is worth.

Reply:
One pattern I've been using is to read examples in acceptance tests: https://github.com/theforeman/puppet-foreman_proxy/blob/d2f07f4a7e460231015f91469f298eb6831c7479/spec/spec_helper_acceptance.rb#L33-L40

It's on my agenda to add that example to voxpupuli-acceptance. The benefit is that you have syntax highlighting and linting for your code samples and you can browse the things we test.

So for now I wouldn't just get rid of it.

Reply:
@ekohl we do the same thing for sensu-puppet

https://github.com/sensu/sensu-puppet/blob/master/spec/spec_helper_acceptance.rb#L46
https://github.com/sensu/sensu-puppet/blob/master/spec/acceptance/examples_spec.rb

Reply:
I would prefer the HEREDOC approach if only to get rid of the example data from the module under `files/`. Really that should only include what is actually used by the module. Given it is named `examples` though, it does seem very clear what is happening. I'll let @dhoppe flip a coin and decide to change it or not :)

Reply:
> @ekohl we do the same thing for sensu-puppet

Glad to hear I'm not alone in this. Given the code is very different, this clearly evolved separate which tells me there's a good chance it's a good idea :)

> I would prefer the HEREDOC approach if only to get rid of the example data from the module under `files/`

That was indeed my reason for it.